src/factories/pid/PIDLookupTable_factory.h

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024, Nathan Brei
+// Subject to the terms in the LICENSE file found in the top-level directory.
+
+#pragma once
+
+#include <edm4hep/MCParticleCollection.h>
+#include <edm4eic/ReconstructedParticleCollection.h>
+#include <edm4hep/utils/vector_utils.h>
+
+#include "services/pid_lut/PIDLookupTable_service.h"
+#include "extensions/jana/JOmniFactory.h"
+
+#include <random>
+
+namespace eicrecon {
+
+struct PIDLookupTableConfig {
+    std::string filename;
+    std::string url;
+};
+
+class PIDLookupTable_factory : public JOmniFactory<PIDLookupTable_factory, PIDLookupTableConfig> {
+
+private:
+    PodioInput<edm4hep::MCParticle> m_particles_in {this};
+    PodioOutput<edm4eic::ReconstructedParticle> m_recoparticles_out {this};
+
+    ParameterRef<std::string> m_filename {this, "filename", config().filename, "Relative to current working directory"};
+    ParameterRef<std::string> m_url {this, "url", config().url, "Only used if not found via filename"};
+    Service<PIDLookupTable_service> m_lut_svc {this};
+
+    std::mt19937 m_gen;
+    std::uniform_real_distribution<double> m_dist {0, 1};
+    const PIDLookupTable* m_lut;
+
+public:
+    void Configure() {
+        m_lut = m_lut_svc().FetchTable(m_filename(), m_url());
+    }
+
+    void ChangeRun(int64_t run_number) {
+    }
+
+    void Process(int64_t run_number, uint64_t event_number) {
+
+        // TODO: This is all very handwavy because I haven't been attending the PID datamodel discussions.
+        // Please look over this carefully and correct as needed!
+
+        for (const auto& mcparticle : *m_particles_in()) {
+
```

Comment:
Creating ReconstructedParticles from MCParticles is not exactly what we want. This only gives an empty shell of a ReconstructedParticle without any tracking or energy information, no associations, etc. We need to 'inject' the PDG information in the ReconstructedParticles that are already created from clustering and tracking, in or immediately after the MatchClusters algorithm, most likely.

Reply:
Do you know what the collection names are for the ReconstructedParticles coming off of the DIRC, pfRICH, and TOF? Or do such collections even exist? It looks like MatchClusters_factory is only matching against ECalClusters at the moment.  MatchCluster's input is "ReconstructedChargedParticles" (which looks like it includes everything from all detectors) and it's output is "ReconstructedParticles" (which isn't named as if it is meant to be ECal-specific)